File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first = float("inf")
        second = float("inf")

        for i in nums:
            if i <= first:
                first = i
            elif i <= second:
                second = i
            else:
                return True
            
        return False

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.go
func increasingTriplet(nums []int) bool {
	one, two := math.MaxInt, math.MaxInt
	for _, v := range nums {
		if v <= one {
			one = v
		} else if v <= two {
			two = v
		} else {
			return true
		}
	}

	return false
}
